packages/editor-ui/src/components/TextWithHighlights.vue

```diff
@@ -0,0 +1,39 @@
+<script lang="ts" setup>
+import type { PropType } from 'vue';
+import { GenericValue } from 'n8n-workflow';
+
+const props = defineProps({
+	content: {
+		type: [Object, String, Number] as PropType<GenericValue>,
+	},
+	search: {
+		type: String,
+	},
+});
+
+const splitTextBySearch = (text = '', search = ''): { tag: 'span' | 'mark'; content: string }[] => {
+	if (!search) {
+		return [
+			{
+				tag: 'span',
+				content: text,
+			},
+		];
+	}
+
+	const pattern = new RegExp(`(${search})`, 'g');
+	const splitText = text.split(new RegExp(pattern, 'gi'));
+
+	return splitText.map((t) => ({ tag: pattern.test(t) ? 'mark' : 'span', content: t }));
+};
+</script>
+
+<template>
+	<span v-if="props.search && typeof props.content === 'string'">
+		<template v-for="part in splitTextBySearch(props.content, props.search)">
```

Comment:
Would it be better to use computed prop instead of a direct method call here, just to avoid redundant reactivity?

Reply:
updated